js/main.js

```diff
@@ -4,21 +4,54 @@
 var $photoDisplay = document.querySelector('#photo-display');
 var $urlInput = document.querySelector('#url');
 var $form = document.querySelector('#input-form');
-var $entries = document.querySelector('ul.container');
+var $entriesContainer = document.querySelector('ul.container');
 var $tabsContainer = document.querySelector('#tabs');
 var $tab = document.querySelectorAll('.tab');
 var $view = document.querySelectorAll('.view');
 var $newEntryButton = document.querySelector('#new-entry-button');
+var editedEntryId = 0;
+var $newEntryHeading = document.querySelector('#new-entry-heading');
+var $editEntryHeading = document.querySelector('#edit-entry-heading');
 
 $urlInput.addEventListener('input', updatePhoto);
 $form.addEventListener('submit', saveEntry);
 document.addEventListener('DOMContentLoaded', displayingPreviousEntry);
 $tabsContainer.addEventListener('click', switchTab);
-$newEntryButton.addEventListener('click', switchToForm);
+$newEntryButton.addEventListener('click', newForm);
+$entriesContainer.addEventListener('click', editEntry);
+
+function editEntry(event) {
+  if (!event.target.hasAttribute('data-edit-id')) {
+    return;
+  }
+  $newEntryHeading.classList.add('hidden');
+  $editEntryHeading.classList.remove('hidden');
+  for (var entryIndex = 0; entryIndex < data.entries.length; entryIndex++) {
+    if (Number.parseInt(event.target.getAttribute('data-edit-id')) === Number.parseInt(data.entries[entryIndex].entryId)) {
+      data.editing = data.entries[entryIndex];
+      break;
+    }
+  }
+  editedEntryId = event.target.getAttribute('data-edit-id');
+  switchToForm(data.editing);
+}
 
 function saveEntry(event) {
   event.preventDefault();
   var obj = {};
+  if (data.editing !== null) {
+    obj.title = $form.elements.title.value;
+    obj.photoURL = $form.elements.url.value;
+    obj.notes = $form.elements.note.value;
+    obj.entryId = editedEntryId;
+    displayNewEntry(obj);
+    $editEntryHeading.classList.add('hidden');
+    $newEntryHeading.classList.remove('hidden');
+    data.entries[editedEntryId - 1] = obj;
```

Comment:
Oh noooo, you should not use ids like this. When you add the ability to delete entries, this will break lots of stuff.